extension/src/Xliff/XLIFFCache.ts

```diff
@@ -0,0 +1,78 @@
+import { Xliff } from "./XLIFFDocument";
+import * as path from "path";
+import { InvalidXmlError } from "../Error";
+import { logger } from "../Logging/LogHelper";
+import * as SettingsLoader from "../Settings/SettingsLoader";
+
+class XliffCache {
+  private cache: Map<string, Xliff>;
+  private enabled: boolean;
+
+  public get size(): number {
+    return this.cache.size;
+  }
+
+  public get isEnabled(): boolean {
+    return this.enabled;
+  }
+
+  constructor() {
+    this.cache = new Map();
+    this.enabled = SettingsLoader.getSettings().enableXliffCache;
+  }
+
+  get(filePath: string): Xliff {
+    const fileName = path.basename(filePath);
+    const xliffDocument = this.isCached(filePath)
+      ? this.cache.get(fileName)
+      : this.read(filePath);
+
+    if (xliffDocument) {
+      if (this.enabled) {
+        this.cache.set(fileName, xliffDocument);
+      }
+      return xliffDocument;
+    }
+    throw new Error(`${fileName} not found.`);
+  }
+
+  private read(filePath: string): Xliff {
+    try {
+      return Xliff.fromFileSync(filePath);
+    } catch (error) {
+      logger.error(
+        `Error while reading "${path.basename(filePath)}":`,
+        (error as Error).message
+      );
+      throw error;
```

Comment:
Sorry, after some more thinking... I don't think that above errors comes from this function, I think they comes from `WorkspaceFunctions.getLangXlfFiles()`
Solve it if you want, otherwise we can look at this later.

Reply:
Looking at `getLangXlfFiles` it doesn't seem particularly prone to bugs. It would seem that the user is running without any localization files. I guess most of them are from english speaking countries?  

We could add:
- Test that `settings.translationFolderPath` exist.
- Have `getLangXlfFiles` return an empty array and handle the length check upstream.
- Change the return signature to take `translationFolderPath` and `appName` which would make it more flexible.

If there's more than that or something feels risky maybe we should handle this in a new issue. Feels like a task that could need some thoughtful designing.

Reply:
I think we should move this to a new issue. 

I believe that we should catch different (more or less expected) errors in `getLangXlfFiles` and handle them in `getHoverText()`.

I think that the scenarios when there are no `translationFolderPath` or it is empty are for PTEs without translations and maybe early in development when there is no translations added yet.